glossary.rst

```diff
@@ -0,0 +1,46 @@
+.. _glossary:
+
+========
+Glossary
+========
+
+This page contains definitions for Flux-specific terms and concepts mentioned
+throughout the various guides and docs on our sites.
+
+------------------------------------------------------------------------------
+
+**Flux Broker**
+  A distributed message broker daemon that provides communications services
+  within a Flux instance.
+
+**Flux Instance**
+  A group of flux-broker processes which are launched via any parallel launch
+  utility that supports PMI (Process Manager Interface).
+
+**Hostlist**
+  A format supported by existing HPC tools to create a compact list of hosts by
+  name, consisting of a common prefix, a number, and an optional suffix.
+
+**Jobspec**
+  A file which defines the resources, tasks, and other attributes of a single
+  program.
+
+**Jobtap Plugin**
+  A built-in or external plugin that can be loaded into the job manager broker
+  module.
+
+**KVS**
+  A component of a Flux instance where information pertaining to the instance,
+  such as resource information, is stored in a key-value format.
+
+**Limits**
+  Maximum values that are enforced across jobs, queues, and users, such as time
+  and/or resource limits.
+
+**PMI**
```

Comment:
What does PMI mean? Why do we use it?

Reply:
I should spell this out too (PMI stands for (I believe) **Process Manager Interface**). This definition is also something I think I could use some help on from the flux-core guys ^

Reply:
Maybe add:
> PMI traditionally provides a key-value store with all-to-all style synchronization to support the bootstrap parallel programs such as MPI.   During wire up, each parallel task can bind to a network endpoint, put its network address into the PMI KVS, synchronize, and then get the addresses of its network peers.  All without requiring the process manager (e.g. Flux) to know anything about the network interconnect hardware.  Brilliant!
>
> Although there are several variations of PMI, Flux implements the original one from Argonne National Lab which is  documented in [RFC 13](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_13.html).   In addition to providing PMI service to the programs it launches, Flux also _uses_ PMI for its own wire-up when it is launched by Flux or a foreign launcher.
>
> PMI was an unfortunate three letter acronym choice, given that MPI was already a thing.  This is known.